### PR TITLE
Improved handleChangeRef test

### PR DESCRIPTION
**Pull request by keiya01**
- added test with `onChange` mode and `onBlur` mode
- replaced from `findByRole` to `waitFor` or async `act`
- added test with `watch`

### CODE REVIEW

File: src/useForm.test.tsx [hunk 1]
```diff
@@ -2046,47 +2050,40 @@ describe('useForm', () => {
           target: { name: 'test', value: 'test' },
         });
 
-        fireEvent.click(screen.getByRole('button'));
-
-        expect((await screen.findByRole('alert')).textContent).toBe('');
-
-        fireEvent.input(screen.getByRole('textbox'), {
-          target: { name: 'wrongName', value: '' },
+        await actComponent(async () => {
```
Comment (bluebill1049) [inline src/useForm.test.tsx:2053]: what's the difference with act and actComponent here?

**Comment by bluebill1049** [comment]
image

File: src/useForm.test.tsx (hunk 1 above)
Comment (keiya01) [reply src/useForm.test.tsx:2053]: `act` is exported from `@testing-library/react-hooks`.
`actComponent` is exported from `@testing-library/react` and renaming from `act` to `actComponent`.

**Comment by keiya01** [comment]
That's horrible lol 😇

File: src/useForm.test.tsx (hunk 1 above)
Comment (bluebill1049) [reply src/useForm.test.tsx:2053]: maybe we should use the same one?

**Comment by bluebill1049** [comment]
> That's horrible lol 😇

that meme remind me of you haha. (opposite)

File: src/useForm.test.tsx (hunk 1 above)
Comment (keiya01) [reply src/useForm.test.tsx:2053]: When you use `render` and `act` that is imported from `@testing-library/react-hooks`, you will received error about act.
Probably `act` can not use with `render` and `actComponent` can not use with `renderHook`.

Comment (bluebill1049) [reply src/useForm.test.tsx:2053]: oh i see, that's fine for now then 👍

**Review by bluebill1049 — APPROVED** [review]

**Comment by keiya01** [comment]
@bluebill1049 
Thank you for reviewing :)

**Comment by bluebill1049** [comment]
🙏 @keiya01 can you please take a look at the other PR?

**Comment by keiya01** [comment]
@bluebill1049 
I'm sorry, I could not review 🙇‍♂️
I will check PR.

**Pull request merged**